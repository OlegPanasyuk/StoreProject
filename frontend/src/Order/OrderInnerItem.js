import React, { Component } from 'react';
import { Card } from 'react-bootstrap';
import PropTypes from 'prop-types';


export class OrderInnerItem extends Component {
    render() {
        let { obj } = this.props;
        return (

            <Card className={'p-0 '}>
                <Card.Header>
                    <Card.Text>
                        {obj.name}
                    </Card.Text>
                </Card.Header>
                <Card.Body>
                    <Card.Text>
                        Price was:{obj.price}
                    </Card.Text>
                </Card.Body>
            </Card>

        );
    }
}

OrderInnerItem.propTypes = {
    obj: PropTypes.object
};

export default OrderInnerItem;