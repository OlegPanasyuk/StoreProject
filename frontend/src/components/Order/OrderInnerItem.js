import React from 'react';
import { Card } from 'react-bootstrap';
import PropTypes from 'prop-types';

export function OrderInnerItem({
    obj
}) {
    return (
        <Card className='p-0'>
            <Card.Header>
                <Card.Text>
                    {obj.name}
                </Card.Text>
            </Card.Header>
            <Card.Body>
                <Card.Text>
                    Price was:
                    {obj.price}
                </Card.Text>
            </Card.Body>
        </Card>
    );
}


OrderInnerItem.propTypes = {
    obj: PropTypes.object
};

OrderInnerItem.defaultProps = {
    obj: {}
};

export default OrderInnerItem;
