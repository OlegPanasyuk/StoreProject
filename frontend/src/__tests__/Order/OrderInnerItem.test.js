import React from 'react';
import OrderInnerItem from '../../Order/OrderInnerItem';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    obj: {
        name: 'admin',
        price: 1222
    }
}) {
    let component = shallow(<OrderInnerItem {...props} />);
    return { component, props };
}

describe('OrderInnerItem', () => {
    it('Component should be rendered', () => {
        let { component } = setupComponent();
        expect(OrderInnerItem.propTypes.obj).toBe(PropTypes.object);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});