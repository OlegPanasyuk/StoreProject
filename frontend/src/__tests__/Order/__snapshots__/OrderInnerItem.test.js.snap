// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderInnerItem Component should be rendered 1`] = `
<Card
  as="div"
  body={false}
  className="p-0 "
>
  <CardHeader>
    <CardText>
      admin
    </CardText>
  </CardHeader>
  <CardBody>
    <CardText>
      Price was:
      1222
    </CardText>
  </CardBody>
</Card>
`;
